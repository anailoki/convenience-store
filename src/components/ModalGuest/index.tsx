import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
} from '@mui/material';

interface IModalGuestProps {
  open: boolean;
  setOpen: (open: boolean) => void;
  onDismiss?: () => void;
  onConfirm?: () => void;
}

const ModalGuest = ({
  open,
  setOpen,
  onConfirm,
  onDismiss,
}: IModalGuestProps) => {
  const handleClose = () => {
    setOpen(false);
  };

  const handleConfirm = () => {
    onConfirm?.();
    handleClose();
  };
  const handleDismiss = () => {
    onDismiss?.();
    handleClose();
  };

  return (
    <div>
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby='alert-dialog-title'
        aria-describedby='alert-dialog-description'
      >
        <DialogTitle id='alert-dialog-title'>
          ¡Regístrate para Guardar Favoritos!
        </DialogTitle>
        <DialogContent>
          <DialogContentText id='alert-dialog-description'>
            Para añadir productos a tu lista de favoritos, necesitas tener una
            cuenta. ¡Regístrate ahora y disfruta de todas las ventajas de ser un
            usuario registrado!
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={handleDismiss}>Seguir como invitado</Button>
          <Button onClick={handleConfirm} autoFocus variant='contained'>
            Registrarse
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};

export default ModalGuest;
