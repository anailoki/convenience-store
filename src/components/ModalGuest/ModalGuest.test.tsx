import { describe, it, expect, vi } from 'vitest';
import { render } from '@testing-library/react';
import ModalGuest from '.';

describe('Render Modal Guest component', () => {
  it('should render close collapse', () => {
    const onClickMock = vi.fn();

    const { getByText } = render(
      <ModalGuest
        title='Modal title'
        description='Modal description'
        open
        setOpen={onClickMock}
      />
    );
    getByText('Modal title');
    getByText('Modal description');
  });

  it('should render the children when collapse is open', async () => {
    const onClickMock = vi.fn();

    const { container } = render(
      <ModalGuest
        title='Modal title'
        description='Modal description'
        open
        setOpen={onClickMock}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render two buttons', () => {
    const onClickMock = vi.fn();

    const { getAllByRole } = render(
      <ModalGuest
        title='Modal title'
        description='Modal description'
        open
        setOpen={onClickMock}
      />
    );

    const buttons = getAllByRole('button');
    expect(buttons.length).toBe(2);
  });

  it('should not render modal', () => {
    const onClickMock = vi.fn();

    const { queryByText } = render(
      <ModalGuest
        title='Modal title'
        description='Modal description'
        open={false}
        setOpen={onClickMock}
      />
    );

    expect(queryByText('Modal title')).toBeNull();
  });
});
