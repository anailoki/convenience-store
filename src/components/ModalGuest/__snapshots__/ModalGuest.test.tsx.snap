// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Render Modal Guest component > should render the children when collapse is open 1`] = `
<div
  aria-hidden="true"
>
  <div />
</div>
`;
